llama-index-integrations/readers/llama-index-readers-zyte-serp/llama_index/readers/zyte_serp/base.py

```diff
@@ -0,0 +1,82 @@
+from typing import Literal, Optional
+from pydantic import Field
+
+from llama_index.core.readers.base import BasePydanticReader
+from llama_index.core.schema import Document
+
+
+class ZyteSerpReader(BasePydanticReader):
+    """Get google search results URLs for a search query.
+
+    Args:
+        api_key: Zyte API key.
+        extract_from: Determines the mode while extracting the .
+            It can take one of the following values: 'html', 'html-text', 'article'
+
+    Example:
+        .. code-block:: python
+
+            from llama_index.readers.zyte_serp import ZyteSerpReader
+
+            reader = ZyteSerpReader(
+               api_key="ZYTE_API_KEY",
+            )
+            docs = reader.load_data(
+                "search query",
+            )
+
+    Zyte-API reference:
+        https://www.zyte.com/zyte-api/
+
+    """
+
+    client: Optional[object] = Field(None)
+    api_key: str
+    extract_from: str
+
+    def __init__(
+        self,
+        api_key: str,
+        extract_from: Literal["httpResponseBody", "browserHtml"] = "httpResponseBody",
```

Comment:
What do you think about leaving the default empty? I think it should be more future-proof, normally we should have the default on zyte api side which just works (currently it's httpResponseBody)

Reply:
Makes sense, done as part of this: https://github.com/rakeshmehta0308/llama_index/pull/2/commits/63ecfd4e4e437e007edc94d5bca05fa38b6a354a